camelot/core/item_model/list_proxy.py

```diff
@@ -142,9 +142,16 @@ def index(self, obj):
             return self._indexed_objects[obj]
         except KeyError:
             i = self._objects.index(obj)
-            # the object is in _objects, but has not been indexed yet,
-            # so index it
+            
+            # The object is present in _objects, but has not been indexed yet, so index it.
+            if i in self._indexed_objects:
+                # If the object's index is already present in the indexed_objects, pop the old existing object at the index,
```

Comment:
ik weet eigenlijk niet of dit een juiste oplossing is.  het gevolg hiervan is dat de index gebruikt in de ui niet meer overeenkomt met de index hier, met potentieel andere gevolgen.  eg : een geselecteerde rij in de ui wijst naar een ander object aan de model kant.

ik vermoed dat in de particular case hier dit geen kwaad kan, omdat de agreement fiche toch volledig gerefresht w.

2 alternatieve zienswijzen : 

- we steken het object in een 'verse index' (len+1 of zo), in principe hoeft de index index in de proxy nt overeen te komen met de index in de onderliggende collectie (dat is bij de query proxy ook niet zo), en dat is eigenlijk by design, want de doel vd proxy is om een 'consistente index' naar de ui toe te hebben.
- we markeren de proxy als 'out of date', bij een eerstvolgende gebruik door de ui van de proxy, moet die dan de proxy wegsmijten, en een volledige refresh doen.

Reply:
Hmm, idd. Dan lijkt me een nieuwe index niet verkeerd,, maar dan moet dat toch gewoon self._length zijn voor de volgende index, en niet length+1? Of zie ik dat verkeerd?

Reply:
dat zie je denk ik juist 👍

Reply:
Ok, zie 0db11ca